tests/Oryx.Integration.Tests/PythonEndToEndTests.cs

```diff
@@ -3,61 +3,81 @@
 // Licensed under the MIT license.
 // --------------------------------------------------------------------------------------------
 
+using Microsoft.Oryx.BuildScriptGenerator.Python;
+using Microsoft.Oryx.Common;
+using Microsoft.Oryx.Tests.Common;
 using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Threading.Tasks;
-using Microsoft.Oryx.Tests.Common;
 using Xunit;
 using Xunit.Abstractions;
 
 namespace Microsoft.Oryx.Integration.Tests
 {
-    [Trait("category", "python")]
-    public class PythonEndToEndTests : PlatformEndToEndTestsBase
+    public abstract class PythonEndToEndTestsBase : PlatformEndToEndTestsBase
     {
-        private const int ContainerPort = 3000;
-        private const string DefaultStartupFilePath = "./run.sh";
+        protected const int ContainerPort = 3000;
+        protected const string DefaultStartupFilePath = "./run.sh";
 
-        private readonly ITestOutputHelper _output;
-        private readonly string _hostSamplesDir;
-        private readonly string _tempRootDir;
+        protected readonly ITestOutputHelper _output;
+        protected readonly string _hostSamplesDir;
+        protected readonly string _tempRootDir;
 
-        public PythonEndToEndTests(ITestOutputHelper output, TestTempDirTestFixture testTempDirTestFixture)
+        public PythonEndToEndTestsBase(ITestOutputHelper output, TestTempDirTestFixture testTempDirTestFixture)
         {
             _output = output;
             _hostSamplesDir = Path.Combine(Directory.GetCurrentDirectory(), "SampleApps");
             _tempRootDir = testTempDirTestFixture.RootDirPath;
         }
+    }
+
+    [Trait("category", "python")]
+    public class PythonEndToEndTests_BackCompat : PythonEndToEndTestsBase
+    {
+        public PythonEndToEndTests_BackCompat(ITestOutputHelper output, TestTempDirTestFixture testTempDirTestFixture)
+            : base(output, testTempDirTestFixture)
+        {
+        }
 
         [Fact]
-        public async Task CanBuildAndRunPythonApp_UsingPython27_AndExplicitOutputStartupFile()
+        public async Task CanRunPythonApp_UsingEarlierBuiltPackagesDirectory()
         {
+            // This is AppService's scenario where previously built apps can still run
+            // fine.
+
             // Arrange
-            var appName = "python2-flask-app";
+            var appName = "flask-app";
             var hostDir = Path.Combine(_hostSamplesDir, "python", appName);
             var volume = DockerVolume.CreateMirror(hostDir);
             var appDir = volume.ContainerDir;
-            var startupFile = "/tmp/startup.sh";
-            var script = new ShellScriptBuilder()
+            var virtualEnvName = "antenv";
+            var buildScript = new ShellScriptBuilder()
+                // App should run fine even with manifest file not present
+                .AddCommand($"oryx build {appDir} -p packagedir={PythonConstants.DefaultTargetPackageDirectory}")
```

Comment:
@gpcastro This test simulates our yesterday's talk. This app is built with package directory and when running a virtual environment name is always provided at line 64